### PR TITLE
CO-1177 There is no more empty PDF in communication for LSV and Direct Debit payment mode

### PR DESCRIPTION
**Pull request by cmtCousteau**

### CODE REVIEW

File: partner_communication_switzerland/models/partner_communication.py [hunk 1]
```diff
@@ -342,8 +344,17 @@ def _get_new_dossier_attachments(self, correspondence=True, payment=True):
                 'group_id.contract_ids').filtered(
                 lambda s: s.state == 'active')
 
+        make_payment_pdf = True
+        for sponsorship in sponsorships:
```
Comment (ecino) [inline partner_communication_switzerland/models/partner_communication.py]: The logic of your code is working and seems good. However you can take advantage of the recordsets methods in order to avoid the loop and all these lines of code. I encourage you to read the [Odoo doc about recordsets](https://www.odoo.com/documentation/10.0/reference/orm.html#recordsets) (especially the mapped function which is very powerful) in order to know them well.

**Review by ecino — CHANGES_REQUESTED** [review]

File: partner_communication_switzerland/models/partner_communication.py [hunk 2]
```diff
@@ -333,6 +333,8 @@ def _get_new_dossier_attachments(self, correspondence=True, payment=True):
         self.ensure_one()
         attachments = OrderedDict()
         report_obj = self.env['report']
+        recurring_contract_group_obj = self.env['recurring.contract.group']
```
Comment (ecino) [inline partner_communication_switzerland/models/partner_communication.py]: This is never used

File: partner_communication_switzerland/models/partner_communication.py [hunk 3]
```diff
@@ -342,8 +344,18 @@ def _get_new_dossier_attachments(self, correspondence=True, payment=True):
                 'group_id.contract_ids').filtered(
                 lambda s: s.state == 'active')
 
+        make_payment_pdf = True
+
+        groupe_ids = sponsorships.mapped('group_id')
+
+        account_payment_mode_list = account_payment_mode_obj.search(['|', ('name', 'like', 'Direct'),('name', 'like', 'LSV')])
```
Comment (ecino) [inline partner_communication_switzerland/models/partner_communication.py]: Line is too long, and you should add a space between the conditions. Also change "Direct" by "Direct Debit"

Comment (ecino) [reply partner_communication_switzerland/models/partner_communication.py]: You could also rename variable to `lsv_dd_modes`

File: partner_communication_switzerland/models/partner_communication.py [hunk 4]
```diff
@@ -342,8 +344,18 @@ def _get_new_dossier_attachments(self, correspondence=True, payment=True):
                 'group_id.contract_ids').filtered(
                 lambda s: s.state == 'active')
 
+        make_payment_pdf = True
+
+        groupe_ids = sponsorships.mapped('group_id')
```
Comment (ecino) [inline partner_communication_switzerland/models/partner_communication.py]: rename variable to `groups`

File: partner_communication_switzerland/models/partner_communication.py [hunk 5]
```diff
@@ -342,8 +344,18 @@ def _get_new_dossier_attachments(self, correspondence=True, payment=True):
...
+
+        account_payment_mode_list = account_payment_mode_obj.search(['|', ('name', 'like', 'Direct'),('name', 'like', 'LSV')])
+        list_recurring_contract_group = groupe_ids.filtered(lambda r: r.payment_mode_id in account_payment_mode_list)
```
Comment (ecino) [inline partner_communication_switzerland/models/partner_communication.py]: rename to `lsv_dd_groups`

File: partner_communication_switzerland/models/partner_communication.py [hunk 6]
```diff
@@ -342,8 +344,18 @@ def _get_new_dossier_attachments(self, correspondence=True, payment=True):
...
+
+        if list_recurring_contract_group is not None and len(list_recurring_contract_group) == len(groupe_ids):
```
Comment (ecino) [inline partner_communication_switzerland/models/partner_communication.py]: You can simply keep the second condition (no need to test not None)

**Review by ecino — CHANGES_REQUESTED** [review]

File: partner_communication_switzerland/models/partner_communication.py [hunk 7]
```diff
@@ -346,12 +345,13 @@ def _get_new_dossier_attachments(self, correspondence=True, payment=True):
 
         make_payment_pdf = True
 
-        groupe_ids = sponsorships.mapped('group_id')
+        groups = sponsorships.mapped('group_id')
 
-        account_payment_mode_list = account_payment_mode_obj.search(['|', ('name', 'like', 'Direct'),('name', 'like', 'LSV')])
-        list_recurring_contract_group = groupe_ids.filtered(lambda r: r.payment_mode_id in account_payment_mode_list)
+        lsv_dd_modes = account_payment_mode_obj.search(['|', ('name', 'like', 'Direct Debit'),
+                                                                     ('name', 'like', 'LSV')])
```
Comment (ecino) [inline partner_communication_switzerland/models/partner_communication.py]: Line is not correctly indented. Should be aligned with '|'

**Review by ecino — APPROVED** [review]

**Comment by ecino** [comment]
You still have PEP8 errors :
```
partner_communication_switzerland/models/partner_communication.py:353:80: E501 line too long (94 > 79 characters)
partner_communication_switzerland/models/partner_communication.py:354:80: E501 line too long (81 > 79 characters)
partner_communication_switzerland/models/partner_communication.py:355:80: E501 line too long (84 > 79 characters)
```
Also you must rebase:
```
git fetch compassion
git rebase compassion/10.0
git push -f
```
